runtime-light/stdlib/fork/wait-queue-functions.h

```diff
@@ -0,0 +1,132 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2025 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include <chrono>
+#include <cstdint>
+#include <optional>
+
+#include "runtime-common/core/core-types/decl/optional.h"
+#include "runtime-light/coroutine/await-set.h"
+#include "runtime-light/coroutine/task.h"
+#include "runtime-light/stdlib/diagnostics/logs.h"
+#include "runtime-light/stdlib/fork/fork-functions.h"
+#include "runtime-light/stdlib/fork/fork-state.h"
+#include "runtime-light/stdlib/fork/future-queue.h"
+#include "runtime-light/stdlib/fork/wait-queue-state.h"
+
+namespace kphp::forks {
+
+inline kphp::coro::task<std::optional<int64_t>> wait_queue_next(int64_t queue_id, std::chrono::nanoseconds timeout) noexcept {
+  auto& wait_queue_instance_st{WaitQueueInstanceState::get()};
+  auto opt_await_set{wait_queue_instance_st.get_queue(queue_id)};
+  if (!opt_await_set.has_value()) [[unlikely]] {
+    kphp::log::warning("future with id {} doesn't associated with wait queue", queue_id);
+    co_return std::nullopt;
+  }
+
+  auto& await_set{opt_await_set.value().get()};
+  // SAFETY: await_set is stored in WaitQueueInstanceState. If it is removed, it will wake up the coroutine
+  static constexpr auto wait_queue_next_task{
+      [](kphp::coro::await_set<int64_t>& await_set) noexcept -> kphp::coro::task<std::optional<int64_t>> { co_return co_await await_set.next(); }};
+  auto wait_result{co_await kphp::coro::io_scheduler::get().schedule(wait_queue_next_task(await_set), kphp::forks::detail::normalize_timeout(timeout))};
+  if (!wait_result) {
+    co_return std::nullopt;
+  }
+
+  if (auto opt_future{*wait_result}; opt_future.has_value()) {
+    auto opt_info{ForkInstanceState::get().get_info(*opt_future)};
+    kphp::log::assertion(opt_info.has_value());
+    auto& fork_info{(*opt_info)};
+    fork_info.get().awaited = false; // allow further f$wait awaiting on the fork
```

Comment:
May we have an explanatory comment for that? It's not obvious why does it work this way